### ISSUE TITLE
DSiWare Game Injection Issue

### ISSUE THREAD

**Issue by Jhynjhiruu** [Plailect/Guide#1194]
I bought and downloaded Sleep Clock: Record and analyse your sleep patterns (EUR), but when going to finally inject the save savedata.bin in GodMode9, I get an error saying there isn't enough space in the FAT mounted directory. What should I do?

**Comment by Plailect** [comment]
How large is the `public.sav` file?

**Comment by Jhynjhiruu** [comment]
Thr public.sav file says it's 65536kb or whatever, it when I mount it as a FAT directory in GodMode9 it only seems to be 61-something kb.

**Comment by Plailect** [comment]
Hm, this seems to be a bug then. Perhaps @d0k3 has an idea?

**Comment by saibotu** [comment]
I think the problem here is, that the header and file allocation table also take up space from `public.sav`. So save actually has to be a bit bigger than 65536 bytes to hold `savedata.bin`.

**Comment by d0k3** [comment]
@Plailect - not a bug in GM9. Of course you don't have the full size of the image file as file system. The FAT header takes up _at least_ 0.5KiB. The FAT table also take up a considerate amount of space. That stuff taking up ~3KiB is pretty much expected.

**Comment by Jhynjhiruu** [comment]
So...

**Comment by 8bitwalugi** [comment]
Hey I can actually help with this!

I encountered the same issue the other day. However, I had a copy of my DSi's NAND decrypted and extracted, with Sudokuhax on that. Since you install v0 Sudoku over Sleep Clock, I figured I could just copy over the save file from my DSi. It worked, Sudoku uses an 80kb save file and simple replacing the Sleep Clock file worked perfectly, even transferred over no issue.

https://gbatemp.net/threads/looking-for-someone-with-cfw-to-help-me-install-in-melbourne-oz-nn3ds-bought-with-11-4.473170/page-2#post-7372464

I can upload my Sudokuhax (v1.0) save file if you want Plai, or I can inject 1.1 into the public.sav and upload that.

**Comment by Jhynjhiruu** [comment]
1.1 public.sav would be the best thing ever. I can seed a torrent of it if required.

**Comment by 8bitwalugi** [comment]
Sorry, I'd rather wait on Plai's word before going ahead with anything. I don't know how he'd feel about it is all.
Plus, another issue is that it's a European save file. I don't know how well that'd play with other regions.

**Comment by Jhynjhiruu** [comment]
I'm European, so starting with just hosting the EUR save file would work well.

**Comment by Plailect** [comment]
The problem with just copying over a larger `public.sav` file is that, as far as I am aware, the system transfer should be detecting that and not transferring the save. If it works for you that's very odd...

**Comment by Jhynjhiruu** [comment]
I think the thing is it's not a bigger public.sav, because of FAT and stuff, it just seems like it should be. Or something.

**Comment by Jhynjhiruu** [comment]
If I had two of the same 3DSs I would try it.

**Comment by Plailect** [comment]
I'll just adjust the save file size (and recommended games) in the guide to be sure.

**Comment by Jhynjhiruu** [comment]
Would CFW->CFW be OK?

**Comment by Plailect** [comment]
I can't be sure that CFW -> CFW would produce identical results to CFW -> OFW.

**Comment by Jhynjhiruu** [comment]
Hm. Is there anything I can do to test it, besides actually doing it?

**Comment by Plailect** [comment]
I'm just going to remove this game and any others that end up being too small from the instructions, so no need to test really.

**Comment by Jhynjhiruu** [comment]
Welp, that's £1.79 wasted.

**Comment by Plailect** [comment]
You can test it if you want, but I'm fairly certain the larger `public.sav` will not work after transfer and you would have to wait a week :/

**Comment by Jhynjhiruu** [comment]
I'll test it with two people who want to swap 3DSes anyway. My friend and his brother want to swap their 2DS and 3DS.

**Comment by 8bitwalugi** [comment]
I can confirm first hand that the larger save file worked. I took from my own copy of Sudoku v0, not expanding a previous file.
